src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -484,8 +503,10 @@ public int compare(RepairSegment seg1, RepairSegment seg2) {
               (range.get().getStart().compareTo(seg.getStartToken())>=0 || range.get().getEnd().compareTo(seg.getEndToken())<=0)
               ) || !range.isPresent()) // Token range condition
           ){
-        segment = seg;
-        break;
+        if(takeLeadOnSegment(seg.getId())) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1066)